### PR TITLE
EasyOCR: init at 1.6.2

### LINKED ISSUE #195340: easyocr
**Project description**

Ready-to-use OCR with 80+ supported languages and all popular writing scripts including Latin, Chinese, Arabic, Devanagari, Cyrillic and etc. 

**Metadata**

* homepage URL: https://github.com/JaidedAI/EasyOCR
* source URL: https://github.com/JaidedAI/EasyOCR
* license: apache-2.0
* platforms: linux

### PR DESCRIPTION
**Pull request by dit7ya**
###### Description of changes

Resolves #195340

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/EasyOCR/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+, buildPythonPackage
+, fetchFromGitHub
+, hdf5
+, numpy
+, opencv3
+, pillow
+, pyaml
+, pyclipper
+, python-bidi
+, pytorch
```
Comment (applePrincess) [inline pkgs/development/python-modules/EasyOCR/default.nix:11]:
```suggestion
, torch
```
pytorch is aliased to torch.

File: pkgs/development/python-modules/EasyOCR/default.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    torchvision
+    pillow
+    pyaml
+    pyclipper
+    pytorch
+    scipy
+    shapely
+  ];
+
+  doCheck = false; # fails as cannot find correct onnx dep
```
Comment (applePrincess) [inline pkgs/development/python-modules/EasyOCR/default.nix:51]:
```suggestion
  checkInputs = [
    onnx
  ];
```
onnx is available in nixpkgs.

Comment (applePrincess) [reply pkgs/development/python-modules/EasyOCR/default.nix:51]: Oh, available but broken on master.

Comment (applePrincess) [reply pkgs/development/python-modules/EasyOCR/default.nix]: I fixed onnx (NixOS/nixpkgs#197194), so could you include it?

Comment (dit7ya) [reply pkgs/development/python-modules/EasyOCR/default.nix]: Done.

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 196693` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>EasyOCR (python310Packages.EasyOCR)</li>
    <li>python39Packages.EasyOCR</li>
  </ul>
</details>

**Review by applePrincess — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -3873,6 +3873,8 @@ with pkgs;
 
   easycrypt-runtest = callPackage ../applications/science/logic/easycrypt/runtest.nix { };
 
+  EasyOCR = with python3.pkgs; toPythonApplication EasyOCR;
```
Comment (bjornfor) [inline pkgs/top-level/all-packages.nix:3876]: Nit: can we make this top-level attribute name lower-case please?

Rationale: it seems pretty common to have package names in lisp-case, and it makes `nix run nixpkgs#easyocr` work without having to set meta.mainProgram.

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -2937,6 +2937,8 @@ in {
 
   easygui = callPackage ../development/python-modules/easygui { };
 
+  EasyOCR = callPackage ../development/python-modules/EasyOCR { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:2940]:
```suggestion
  easyocr = callPackage ../development/python-modules/easyocr { };
```
python attrs must be normalized

File: pkgs/development/python-modules/EasyOCR/default.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    owner = "JaidedAI";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-f+JBSnFMRvVlhRRiL1rJb7a0CNjZPuh6r8r3K1meQCk=";
+  };
+
+  postPatch = ''
+    substituteInPlace requirements.txt \
+      --replace "opencv-python-headless<=4.5.4.60" ""
+    substituteInPlace requirements.txt \
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/EasyOCR/default.nix:33]:
```suggestion
      --replace "opencv-python-headless<=4.5.4.60" "" \
```

File: pkgs/development/python-modules/EasyOCR/default.nix [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+, torch
+, scikitimage
+, scipy
+, shapely
+, torchvision
+, onnx
+}:
+
+buildPythonPackage rec {
+  pname = "EasyOCR";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/EasyOCR/default.nix:20]:
```suggestion
  pname = "easyocr";
```

File: pkgs/development/python-modules/EasyOCR/default.nix [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  version = "1.6.2";
+
+  src = fetchFromGitHub {
+    owner = "JaidedAI";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/EasyOCR/default.nix:25]:
```suggestion
    repo = "EasyOCR";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Why do we need this in pythonPackages? Is this being used as a depdendency?

**Comment by armijnhemel** [comment]
> Why do we need this in pythonPackages? Is this being used as a depdendency?

It is a Python library for OCRing, so it makes sense to go into pythonPackages.

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.easyocr

**Pull request merged**